app/models/nomis/sentence_detail.rb

```diff
@@ -27,6 +30,20 @@ def conditional_release_date
       @conditional_release_override_date.presence || @conditional_release_date
     end
 
+    def recall_release_date
+      return decide_which_recall_release_date if @actual_parole_date.blank?
+      return @actual_parole_date if recall_release_override_date_is_earlier?
+      decide_which_recall_release_date
+    end
+  
+    def recall_release_override_date_is_earlier?
+      @post_recall_release_override_date.blank? || @post_recall_release_override_date > @actual_parole_date
```

Comment:
I think we're missing a check against – 'is post recall release date earlier than actual parole date'?